jdaviz/configs/default/plugins/plot_options/plot_options.py

```diff
@@ -38,8 +46,31 @@ def _on_select_viewer_message(self, msg):
         # message from elsewhere requesting to change the selected viewer
         self.selected_viewer = msg.viewer
 
+    def _on_data_changed(self):
+        # Make sure new data has it's uncertainty plotted
+        self._toggle_uncertainty(None)
+
     @observe("selected_viewer")
     def _selected_viewer_changed(self, event={}):
         viewer = self.app.get_viewer_by_id(event.get('new', self.selected_viewer))
         self.viewer_widget = viewer.viewer_options
         self.layer_widget = viewer.layer_options
+
+    @observe("show_uncertainty")

```

Comment:
Actually, will such hardcoding break MAST? cc @havok2063

Reply:
To debate with myself, I don't think it'll break MAST though we might need extra logic if MAST wants this feature. Future work!

Reply:
Do you mean the per-config hardcoded viewer IDs?

Reply:
I don't see where the hard-coding is taking place?  Lines 60-66 below? But my answer without context is "it might break mast??".  Currently in our notebook we customize the `settings.configuration` value and call it "mastviz" when we pass it into the main application.  It sounds like that might not be a good idea if internal jdaviz code checks for one of the specific viztools for functionality.    We currently customize config items from `settings`, `toolbar`, and `tray`, but not `viewer_area`.

Reply:
There's no other reason for us renaming the configuration except to explicitly say "this is not exactly the same Cubeviz, Specviz, etc that you'll see in a notebook".  And of course we only say this to people who look at the internal notebook, which is me.

Reply:
It might be best to just loop over all viewers in the app and call `show_uncertainties` on any that are an instance of `SpecvizProfileView`.   The UI logic also needs to be changed from checking the app-config to checking the type of the currently selected viewer (so the switch does not show in the image viewers in cubeviz, etc)

Reply:
I will also change our notebook so it keeps the original configuration name.

Reply:
see #1208 for changing this to per-viewer and removing reliance on the configuration name.